### PR TITLE
Fix typos and small improvements to flow of JOSS paper text

### PR DESCRIPTION
**Pull request by matt-graham**
Some very minor suggested improvements to JOSS paper text as part of review https://github.com/openjournals/joss-reviews/issues/5021

### CODE REVIEW

File: joss/paper.md [hunk 1]
```diff
@@ -29,14 +29,14 @@ bibliography: paper.bib
...
-Across scientific fields, the Bayesian framework is used to account for uncertainties in inference [@ellison2004bayesian; @lancaster2004introduction; @von2011bayesian]. However, for models with more than a few parameters, exact inference is intractable. A frequently used strategy is to approximately sample the posterior distribution on models' parameters with a Markov chain Monte Carlo (MCMC). Yet conventional MCMC methods relying on local updates can take a prohibitive time to converge when posterior distributions have complex geometries (see e.g., [@rubinstein_simulation_2017] for a textbook).
+Across scientific fields, the Bayesian framework is used to account for uncertainties in inference [@ellison2004bayesian; @lancaster2004introduction; @von2011bayesian]. However, for models with more than a few parameters, exact inference is intractable. A frequently used strategy is to approximately sample the posterior distribution on a model's parameters with a Markov chain Monte Carlo (MCMC) method. Yet conventional MCMC methods relying on local updates can take a prohibitive time to converge when posterior distributions have complex geometries (see e.g., @rubinstein_simulation_2017).
```
Comment (matt-graham) [inline joss/paper.md:32]: At the moment there is some switching between plural and singular forms - it should either be
> sample the posterior distributions on models' parameters
or
> sample the posterior distribution on a model's parameters
I've suggested latter (singular) option as think it reads better.

'A Markov chain Monte Carlo' doesn't read right - I think either should be no determiner or use as qualifier for 'method' as suggested.

The `rubinstein_simulation_2017` citation shouldn't be in parenthesis as its referred to directly, and I think 'for a textbook' doesn't scan right and isn't necessary.

File: joss/paper.md [hunk 2]
```diff
@@ -29,14 +29,14 @@ bibliography: paper.bib
...
+`flowMC` is a Python library implementing accelerated MCMC leveraging deep generative modelling as proposed by @Gabrie2022, built on top of the machine learning libraries JAX [@jax2018github] and Flax [@flax2020github]. At its core, `flowMC` uses a combination of Metropolis-Hastings Markov kernels using local and global proposed moves. While multiple chains are run using local-update Markov kernels to generate approximate samples over the region of interest in the target parameter space, these samples are used to train a normalizing flow (NF) model to approximate the samples' density. The NF is then used in an independent Metropolis-Hastings kernel to propose global jumps across the parameter space. The `flowMC` sampler can handle non-trivial geometry, such as multimodal distributions and distributions with local correlations.
```
Comment (matt-graham) [inline joss/paper.md:34]: Typo in spelling of Metropolis and I think 'an' is the correct determiner here as there are multiple possible independent Metropolis-Hastings kernels and this is a single instance.

File: joss/paper.md [hunk 3]
```diff
@@ -29,14 +29,14 @@ bibliography: paper.bib
...
 
 The key features of `flowMC` are summarized in the following list:
 
 - Since `flowMC` is built on top of JAX, it supports gradient-based samplers through automatic differentiation such as the Metropolis-adjusted Langevin algorithm (MALA) and Hamiltonian Monte Carlo (HMC).
-- `flowMC` uses state-of-the-art NF models such as rational quadratic splines (RQS) to power its global sampler. These models are efficient in capturing important features within a relatively short training time.
+- `flowMC` uses state-of-the-art NF models such as rational quadratic splines (RQS) to power its global proposals. These models are efficient in capturing important features within a relatively short training time.
```
Comment (matt-graham) [inline joss/paper.md:39]: Replacing sampler with proposals for consistency with previous language.

File: joss/paper.md [hunk 4]
```diff
@@ -72,17 +72,17 @@ If further sampling from the posterior is necessary, the flow trained during a p
 The mathematical details of the method are explained in [@Gabrie2022; @Gabrie2021a].
 
 ***Use of accelerators***
-`flowMC` is built on top of JAX, that supports the use of GPU and TPU accelerators by default.
+`flowMC` is built on top of JAX, which supports the use of GPU and TPU accelerators by default.
```
Comment (matt-graham) [inline joss/paper.md:75]: Not entirely sure about this one but 'that' doesn't read quite right to me here.

**Comment by kazewong** [comment]
All changes look sensible to me. Merging and closing this PR

**Pull request merged**